str/trtools/merge_str_runner.py

```diff
@@ -95,46 +72,14 @@ def main(
         # read in input file paths
         batch_vcfs = []
         num_samples = 0
-        with to_path(sample_list_1).open() as f_1:
-            ids_1 = [line.strip() for line in f_1]
-            for id in ids_1:
-                each_vcf = os.path.join(
-                    input_dir_1, f'{id}_eh_shard{shard_index}.reheader.vcf.gz'
-                )
-                batch_vcfs.append(
-                    b.read_input_group(
-                        **{
-                            'vcf.gz': each_vcf,
-                            'vcf.gz.tbi': f'{each_vcf}.tbi',
-                        }
-                    )['vcf.gz']
-                )
-            num_samples = num_samples + len(ids_1)
-
-        # if second file directory is specified, read in input file paths:
-        if input_dir_2 is not None:
-            with to_path(sample_list_2).open() as f_2:
-                ids_2 = [line.strip() for line in f_2]
-                for id in ids_2:
-                    each_vcf = os.path.join(
-                        input_dir_2, f'{id}_eh_shard{shard_index}.reheader.vcf.gz'
-                    )
-                    batch_vcfs.append(
-                        b.read_input_group(
-                            **{
-                                'vcf.gz': each_vcf,
-                                'vcf.gz.tbi': f'{each_vcf}.tbi',
-                            }
-                        )['vcf.gz']
-                    )
-                num_samples = num_samples + len(ids_2)
-        # similarly, if third and fourth file directories are specified - read in input file paths:
-        if input_dir_3 is not None:
-            with to_path(sample_list_3).open() as f_3:
-                ids_3 = [line.strip() for line in f_3]
-                for id in ids_3:
+        cpg_ids =[]
+        for input_dir, sample_list in zip(input_file_paths[::2], input_file_paths[1::2]):
```

Comment:
I'm not sure this logic works... in your example command up top you have one continuous string with commas, as
`dir1,sample1,dir2,sample2`. You would need to break that into a list, then iterate over the list in pairs. This logic would do a staggered iteration over a continuous list, but first you need to break it into a list of separate elements.

With `nargs=-1` in your click argument description you can ingest any number of strings into a list, which you can accommodate by making the args `dir1,sample1    dir2,sample2`. If you comma-separate the directory and sample list, then separate each pair with spaces, you end up with

```
script.py ... dir1,sample1    dir2,sample2
...
input_file_paths = ['dir1,sample1', 'dir2,sample2']
for dir, sam in input_file_paths.split(','): 
   ...
```

Alternatively your logic as currently written will work, but you need to remove your commas from the example invocation up top so that each of the strings is kept separately in a continuous list

Reply:
i changed it to follow the args convention of dir1,sample1    dir2,sample2 - pairs are separated by space.